test/minikube/001-kube.bats

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bats
+#
+# Tests of podman kube commands with minikube
+#
+
+load ../system/helpers.bash
+
+function setup(){
+    # only set up the minikube cluster before the first test
+    if [ "$BATS_TEST_NUMBER" -eq 1 ]; then
+        minikube start
+        echo $output >&2
+    fi
+    # wait for a bit so all the services in the cluster come up
+    sleep 10
+}
+
+###############################################################################
+# BEGIN tests
+
+@test "minikube - check cluster is up" {
+    run minikube kubectl get nodes
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"Ready"* ]]
+    run minikube kubectl get pods
+    [ "$status" -eq 0 ]
+    [[ "$output" == "No resources found in default namespace." ]]
+}
+
+@test "minikube - deploy generated container yaml to minikube" {
+    cname="test-ctr"
+    fname="/tmp/f$(random_string 6).yaml"
+    run_podman container create --name $cname $IMAGE top
+    [ "$status" -eq 0 ]
+    run_podman kube generate -f $fname $cname
+    [ "$status" -eq 0 ]
+
+    # deploy to the minikube cluster
+    run minikube kubectl -- apply -f $fname
+    [ "$status" -eq 0 ]
+    [[ "$output" == "pod/$cname-pod created" ]]
+    # need to wait for the pod to be created and started before checking status
+    sleep 5
+    run minikube kubectl get pods
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"Running"* ]]
+
+}
+
+@test "minikube - deploy generated pod yaml to minikube" {
+    pname="test-pod"
+    cname1="test-ctr1"
+    cname2="test-ctr2"
+    fname="/tmp/f$(random_string 6).yaml"
+
+    run_podman pod create --name $pname --publish 9999:8888
+    [ "$status" -eq 0 ]
```

Comment:
Eeeeeek! Please never, ever, ever use this. Although this is normal in BATS-land, it is a nightmare in terms of usability. When it fails, all it tells you is something like "duh, failed" but without any actual useful information.

`run_podman` already does status checks. It is built-in. With a numeric arg, like `run_podman 1`, it will guarantee that exit code is 1, and fail (with an actually useful message) if it isn't.

Reply:
Fixed